pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -296,18 +296,43 @@ cdef class _Timestamp(datetime):
             val = tz_convert_single(self.value, 'UTC', self.tz)
         return val
 
-    cpdef int _get_field(self, field):
+    cdef int _get_field(_Timestamp self, field):
+        # Note: because _get_field is `cdef`, it is not visible in the python
+        # namespace.  Any methods that need to call `_get_field` must be
+        # defined in _Timestamp, not Timestamp.
         cdef:
             int64_t val
             ndarray[int32_t] out
         val = self._maybe_convert_value_to_local()
         out = get_date_field(np.array([val], dtype=np.int64), field)
         return int(out[0])
 
-    cpdef _get_start_end_field(self, field):
+    @property
+    def dayofyear(self):
+        return self._get_field('doy')
+
+    @property
+    def week(self):
+        return self._get_field('woy')
+
+    @property
+    def quarter(self):
+        return self._get_field('q')
+
+    @property
+    def days_in_month(self):
+        return self._get_field('dim')
+
+    cdef bint _get_start_end_field(_Timestamp self, field):
```

Comment:
why does moving this to _Timestamp help

Reply:
> why does moving this to _Timestamp help

Among the todos was getting _get_start_end_field and _get_field out of the user-facing namespace.  This accomplishes that.

It also adds type declarations, so should be marginally more performant.

Reply:
why do we care about ``_get_start_end_field`` in the user namespace, its a private method?

type declarations are fine (pls run an asv to confirm)